com.unity.render-pipelines.core/Runtime/Volume/KeyframeUtility.cs

```diff
@@ -0,0 +1,294 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using Unity.Collections;
+using UnityEngine.Assertions;
+
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// A helper function for interpolating AnimationCurves together. In general, curves can not be directly blended
+    /// because they will have keypoints at different places. InterpAnimationCurve traverses through the keypoints.
+    /// If both curves have a keypoint at the same time, they keypoints are trivially lerped together. However
+    /// if one curve has a keypoint at a time that is missing in the other curve (which is the most common case),
+    /// InterpAnimationCurve calculates a synthetic keypoint at that time based on value and derivative, and interpolates
+    /// the resulting keys.
+    ///
+    /// Note that this function should only be called by internal rendering code. It creates a small pool of animation
+    /// curves and reuses them to avoid creating garbage. The number of curves needed is quite small, since curves only need
+    /// to be used when interpolating multiple volumes together with different curve parameters. The underlying interp
+    /// function isn't allowed to fail, so in the case where we run out of memory we fall back to returning a single keyframe.
+    /// </summary>
+    public class KeyframeUtility
+    {
+        /// <summary>
+        /// Helper function to remove all control points for an animation curve. Since animation curves are reused in a pool,
+        /// this function clears existing keys so the curve is ready for reuse.
+        /// </summary>
+        /// <param name="curve">The curve to reset.</param>
+        static public void ResetAnimationCurve(ref AnimationCurve curve)
```

Comment:
Do you need `ref`? You are not doing any assignment to `curve`. Besides this, I think this method can be directly into trunk,  as it seems helpfull for users.

If we decide to not put it into trunk, I would recommend you to create an extension class for animation curve.

```
namespace UnityEngine
{
    public static class AnimationCurveExtensions
    {
        static public void Reset(this AnimationCurve curve)
        {
            int numPoints = curve.length;
            for (int i = numPoints - 1; i >= 0; i--)
            {
                curve.RemoveKey(i);
            }
        }
     }
}
```

In this way we could do something like `animationCurve.Reset();` and if someday someone creates that reset method it will be easy changed.